x-pack/plugins/observability_onboarding/public/assets/standalone_agent_setup.sh

```diff
@@ -56,13 +56,20 @@ updateStepProgress() {
   local STEPNAME="$1"
   local STATUS="$2" # "incomplete" | "complete" | "disabled" | "loading" | "warning" | "danger" | "current"
   local MESSAGE=${3:-}
+  local PAYLOAD=${4:-}
+  local data=""
+  if [ -z "$PAYLOAD" ]; then
+    data="{\"status\":\"${STATUS}\", \"message\":\"${MESSAGE}\"}"
+  else
+    data="{\"status\":\"${STATUS}\", \"message\":\"${MESSAGE}\", \"payload\":${PAYLOAD}}"
```

Comment:
I am not 100% sure, but back of mind, the $PAYLOAD must also be surrounded with escaped double quotes no ?

```suggestion
      data="{\"status\":\"${STATUS}\", \"message\":\"${MESSAGE}\", \"payload\":\"${PAYLOAD}\"}"
```